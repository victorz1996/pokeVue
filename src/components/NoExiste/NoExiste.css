.botonNoExiste {
    background: #f22539;
    border-radius: 10rem;
    color: aliceblue;
    height: 40px;
    width: 150px;
}