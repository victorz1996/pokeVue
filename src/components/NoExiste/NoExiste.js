/**
 * @namespace NoExiste
 * @description Componente que contiene la informacion que se muestra cuando no
 * existen coincidencias en la busqueda
 * @author       Victor Zabala <victorma164@gmail.com>
 */
export default {
    
    methods: {
        /**
        * @memberOf NoExiste
        * @function volverInicio Volver al inicio
        * @description Metodo que se sirve para volver al inicio de la aplicacion
        */
        volverInicio(){
            this.$emit('volverInit', true)
        }
    },
}