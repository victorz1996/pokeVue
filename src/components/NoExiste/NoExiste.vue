<template>
  <div class="mt-4">
    <h2><strong>Uh-oh!</strong></h2>
    <h6>You look lost on your journey!</h6>
    <button @click="volverInicio" class="btn botonNoExiste mt-3">Go back home</button>
  </div>
</template>
<script src="./NoExiste.js"></script>
<style lang="css" src="./NoExiste.css"></style>