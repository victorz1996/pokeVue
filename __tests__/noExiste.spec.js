import { shallowMount } from '@vue/test-utils'
import NoExiste from "../src/components/NoExiste/NoExiste.vue"

let cmp

    // eslint-disable-next-line no-undef
    beforeEach(() => {
      cmp = shallowMount(NoExiste)
    })
    // eslint-disable-next-line no-undef
    describe('Componente NoExiste', () => {
      // eslint-disable-next-line no-undef
      test('Instancia de Vue.js',() => {
        // eslint-disable-next-line no-undef
        expect(cmp.isVueInstance).toBeTruthy()
      })

      // eslint-disable-next-line no-undef
      it('HTMl renderizado structure esperada',()=>{
        // eslint-disable-next-line no-undef
        expect(cmp.element).toMatchSnapshot()
    })
})