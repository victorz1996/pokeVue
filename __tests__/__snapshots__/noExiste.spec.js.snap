// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente NoExiste HTMl renderizado structure esperada 1`] = `
<div
  class="mt-4"
>
  <h2>
    <strong>
      Uh-oh!
    </strong>
  </h2>
   
  <h6>
    You look lost on your journey!
  </h6>
   
  <button
    class="btn botonNoExiste mt-3"
  >
    Go back home
  </button>
</div>
`;
